### PR TITLE
Add guideline for maintaing a clean Git history with squash merge cases

### PR DESCRIPTION
**Pull request by csarven**

### CODE REVIEW

**Review by VirginiaBalseiro — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -192,6 +192,10 @@ repository, create a PR.
   (after two empty lines). See [github
   tutorial](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors).
 
+* To have a maintain a clean Git history, consider using [squash
```
Comment (VirginiaBalseiro) [inline CONTRIBUTING.md:195]:
```suggestion
* To help maintain a clean Git history, consider using [squash
```

**Review by VirginiaBalseiro — APPROVED** [review]

**Comment by elf-pavlik** [comment]
GitHub docs warn:
> You lose information about when specific changes were originally made and who authored the squashed commits.

Given that, should we add more detailed guidance for when to squash and not to squash?

I think GitHub will still add `Co-authored-by` but even if someone just merges they will appear as a full co-author.

When someone adds a commit correcting their own previous commit, we could also follow approach of [Fixing commits with git commit --fixup and git rebase --autosquash](https://jordanelver.co.uk/blog/2020/06/04/fixing-commits-with-git-commit-fixup-and-git-rebase-autosquash/)

**Comment by csarven** [comment]
I'm not sure how much detail we should introduce here about merging, squashing, rebasing. We could link out to best practices perhaps? Perhaps a one-liner for squash and rebase?

I didn't want to overdo it here, and just wanted to jot down a general *when* to squash:

>especially when incorporating reviews and code additions.

If "one concern, one PR" is followed, squash will be adequate. For a "large" concern with atomic commits, rebasing makes sense - perhaps we say that exactly? (I'm not a fan of having merge commits in the history, and I'm not sure if merge commits are needed.)

(There are separate cases with Co-authored-by, e.g.,  it would be a nice practice to attribute the scribes as the co-authors of the commit even when they're not making the PR, or in other cases where we make sure the original authors are noted.)

**Comment by VirginiaBalseiro** [comment]
Merged as it has been discussed in meetings and there have been no objections. We can make improvements in separate PRs.

**Pull request merged**